library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java

```diff
@@ -818,6 +818,12 @@ protected void releaseSourceInternal() {
         }
       }
     }
+    boolean isRedirect = !loadable.getUri().equals(this.manifestUri);
```

Comment:
As per my comment in https://github.com/google/ExoPlayer/issues/6907, this change will need updating to let `manifest.location` take priority. You should also do the `isSameUriInstance` check as above. So I think you probably want to replace L810 to L826 with something that looks more like (disclaimer: untested):

```
Uri newManifestUri = manifest.location != null ? manifest.location : loadable.getUri();
synchronized (manifestUriLock) {
  // This condition checks that replaceManifestUri wasn't called between the start and end of
  // this load. If it was, we ignore the manifest location and prefer the manual replacement.
  @SuppressWarnings("ReferenceEquality")
  boolean isSameUriInstance = loadable.dataSpec.uri == manifestUri;
  if (isSameUriInstance) {
    manifestUri = newManifestUri;
  }
}
```